cms/static/js/views/instructor_info.js

```diff
@@ -0,0 +1,100 @@
+// Backbone Application View: Instructor Information
+
+define([  // jshint ignore:line
+        'jquery',
+        'underscore',
+        'backbone',
+        'gettext',
+        'js/utils/templates',
+        "js/models/uploads",
+        "js/views/uploads"
+    ],
+    function ($, _, Backbone, gettext, TemplateUtils, FileUploadModel, FileUploadDialog) {
+        'use strict';
+        var InstructorInfoView = Backbone.View.extend({
+
+             events : {
+                 'click .remove-instructor-data': 'removeInstructor',
+                 'click .action-upload-instructor-image': "uploadInstructorImage"
+             },
+
+            initialize: function(options) {
+                // Set up the initial state of the attributes set for this model instance
+                 _.bindAll(this, 'render');
+                this.template = this.loadTemplate('course-instructor-details');
+                this.model = options.model;
+            },
+
+            loadTemplate: function(name) {
+                // Retrieve the corresponding template for this model
+                return TemplateUtils.loadTemplate(name);
+            },
+
+            render: function() {
+                // Assemble the render view for this model.
+                $("span.course-instructor-details-fields").empty();
+                var self = this;
+                var instructors = this.model.get('instructor_info').instructors;
+                $.each(instructors, function( index, data ) {
+                    $(self.el).append(self.template({
+                        data: data,
+                        index: index,
+                        instructors: instructors.length
+                    }));
+                });
+
+                // Avoid showing broken image on mistyped/nonexistent image
+                this.$el.find('img').error(function() {
+                    $(this).hide();
+                });
+                this.$el.find('img').load(function() {
+                    $(this).show();
+                });
+            },
+
+            removeInstructor: function(event) {
+                /*
+                 * Remove course Instructor fields.
+                 * */
+                event.preventDefault();
+                var index = event.currentTarget.getAttribute('data-index'),
+                    existing_info = _.clone(this.model.get('instructor_info'));
+                existing_info.instructors.splice(index, 1);
+                this.model.set('instructor_info', existing_info);
+                this.model.trigger("change:instructor_info", this.model );
```

Comment:
Change events should automatically be triggered on save. Why do you need to explicitly trigger an event?

http://backbonejs.org/#Model-set